services/galley/test/integration/API/Teams/LegalHold.hs

```diff
@@ -395,39 +394,6 @@ testNoConsentCannotBeInvited = do
     API.Util.postQualifiedMembers userLHNotActivated (Qualified peer2 localdomain :| []) qconvId
       >>= errWith 403 (\err -> Error.label err == "missing-legalhold-consent")
 
-testCannotCreateGroupWithUsersInConflict :: HasCallStack => TestM ()
-testCannotCreateGroupWithUsersInConflict = do
-  -- team that is legalhold whitelisted
-  (legalholder :: UserId, tid) <- createBindingTeam
-  userLHNotActivated <- (^. Team.userId) <$> addUserToTeam legalholder tid
-  putLHWhitelistTeam tid !!! const 200 === statusCode
-
-  -- team without legalhold
-  (peer :: UserId, teamPeer) <- createBindingTeam
-  peer2 <- (^. Team.userId) <$> addUserToTeam peer teamPeer
-
-  do
-    postConnection userLHNotActivated peer !!! const 201 === statusCode
-    void $ putConnection peer userLHNotActivated Conn.Accepted <!! const 200 === statusCode
-
-    postConnection userLHNotActivated peer2 !!! const 201 === statusCode
-    void $ putConnection peer2 userLHNotActivated Conn.Accepted <!! const 200 === statusCode
```

Comment:
do we know why this works? I don't get it

Reply:
I am not sure what exactly you're talking about, but maybe I can help with this. Can you clarify what you mean by "this"?

Reply:
We’re trying to post a conversation to Alice after only connecting Bob with Charlie and Debora (see new code, there it’s more readable), the new code errors saying Bob is not connected (to Alice) which makes sense but I don’t get why it doesn’t in the old test? Perhaps in the old test team members were somehow implicitly connected already?

Reply:
the reason is, that even though `alice` and `bob` are automatically connected, because they're in a team, you still have to specify the team the conversation belongs to, s.t. they count as connected; I don't quite understand what happens if bob and alice each join a conversation that was created of a foreign team owner, will this always error with "not connected"

Reply:
I'm sorry, I still don't quite understand the issue. Can you write a test for your question and see what you get in response?

In general, all users within a team are connected by default, without them having to directly establish pair-wise connections.

Your rewrite of the test isn't quite faithful. A regular group conversation is not the same as a team conversation; they follow different rules. In your tests you create a plain group conversation, whereas you should be setting a team as the owner of a conversation. You can do that by setting the `team` parameter of the `defProteus` argument.

Reply:
yes, that was the issue and that's why it didn't work, I found the issue yesterday morning; the problem was that the old combinator set that option kinda implicitly. :)